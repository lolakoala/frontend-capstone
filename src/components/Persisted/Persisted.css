#logo {
  margin-left: 50px;
}

#persisted-user {
  display: flex;
  font-size: 18px;
  top: 20px;
  right: 25%;
}
