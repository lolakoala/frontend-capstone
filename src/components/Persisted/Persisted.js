import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Avatar, RaisedButton } from 'material-ui';
import css from './Persisted.css';
import logo from '../../assets/logo1.png';

class Persisted extends Component {
  panic = () => {
    // use twilio API to text the mental health hotline
  };

  render () {
    const { currentUser, signOut } = this.props;
    const { pathname } = this.props.location;
    const userKeys = Object.keys(currentUser);

    const renderUser = () => {
      if (userKeys.length > 2 && (!pathname.includes('profile') || !pathname.includes('Profile'))) {
        return userInfo;
      }
    };

    const userInfo = <div id="current-user">
      {/* placeholder avatar, should fetch from server?  */}
      <Avatar src="https://speakerdata2.s3.amazonaws.com/photo/image/839843/thats-what-she-said-0413-1-lgn.jpg" alt="Your user photo"/>
      <h3>{currentUser.userName}</h3>
    </div>;

    const signOutButton = <RaisedButton
      label="Sign Out"
      onClick={signOut}
      linkButton={true}
      href="/"
      style={{
        position: 'fixed',
        top: '70px',
        right: '25%'
      }}
    />;

    const dashLink = <RaisedButton
      label="Home"
      id="home-button"
      linkButton={true}
      href="/dash"
      style={{
        position: 'fixed',
        top: '120px',
        right: '25%'
      }}
    />;

    const pathArray = ['/dash', '/', '/createProfile', '/selectCity'];

    return (
      <div>
        <RaisedButton
          label="PANIC"
          onClick={this.panic}
          id="panic-button"
          style={{
            position: 'fixed',
            top: '20px',
            right: '25%'
          }}
        />
        {renderUser()}
        <img src={logo} alt="logo" id="logo"/>
        {pathname !== '/' ? signOutButton : null}
        {!pathArray.includes(pathname) ? dashLink : null}
      </div>
    );
  }
}

export default Persisted;

Persisted.propTypes = {
  currentUser: PropTypes.object,
  location: PropTypes.object,
  signOut: PropTypes.func
};
