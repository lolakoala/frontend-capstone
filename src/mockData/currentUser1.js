const currentUser1 = {
  email: 'lola@gmail.com',
  city: 'Denver',
  userName: 'lolakoala',
  aboutMe: 'Lola loves doggos.'
};

export default currentUser1;
