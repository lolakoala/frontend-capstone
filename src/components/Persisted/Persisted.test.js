import React from "react";
import Persisted from './Persisted.js';
import { shallow, configure } from "enzyme";
import currentUser1 from '../../mockData/currentUser1.js';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });


describe('Persisted', () => {
  const mockFn = jest.fn();
  const wrapper = shallow(<Persisted
    currentUser={currentUser1}
    location={{
      pathname: '/toolbox'
    }}
    signOut={mockFn}
    history={{}}
  />);

  it('should exist', () => {
    expect(wrapper).toBeDefined();
  });

  it('should match snap', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have 3 raised buttons', () => {
    const raisedButton = wrapper.find('RaisedButton');

    expect(raisedButton.length).toEqual(3);
  });

  it('should have a logo', () => {
    const logo = wrapper.find('img');

    expect(logo.length).toEqual(1);
  });

  it('should have an avatar', () => {
    const avatar = wrapper.find('Avatar');

    expect(avatar.length).toEqual(1);
  });

  it('should have a username', () => {
    const userName = wrapper.find('h3');

    expect(userName.length).toEqual(1);
  });

  it('should call function on click of signout', () => {
    const signOut = wrapper.find('RaisedButton').at(1);

    signOut.simulate('click');
    expect(mockFn).toHaveBeenCalledTimes(1);
  });
});
