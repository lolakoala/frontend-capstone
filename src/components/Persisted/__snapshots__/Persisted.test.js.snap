// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Persisted should match snap 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Persisted
    currentUser={
        Object {
            "aboutMe": "Lola loves doggos.",
            "city": "Denver",
            "email": "lola@gmail.com",
            "userName": "lolakoala",
          }
    }
    history={Object {}}
    location={
        Object {
            "pathname": "/toolbox",
          }
    }
    signOut={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <RaisedButton
          disabled={false}
          fullWidth={false}
          id="panic-button"
          label="PANIC"
          labelPosition="after"
          onClick={[Function]}
          primary={false}
          secondary={false}
          style={
                    Object {
                              "position": "fixed",
                              "right": "25%",
                              "top": "20px",
                            }
          }
/>,
        <div
          id="current-user"
>
          <Avatar
                    alt="Your user photo"
                    size={40}
                    src="https://speakerdata2.s3.amazonaws.com/photo/image/839843/thats-what-she-said-0413-1-lgn.jpg"
          />
          <h3>
                    lolakoala
          </h3>
</div>,
        <img
          alt="logo"
          id="logo"
          src="logo1.png"
/>,
        <RaisedButton
          disabled={false}
          fullWidth={false}
          href="/"
          label="Sign Out"
          labelPosition="after"
          linkButton={true}
          onClick={[Function]}
          primary={false}
          secondary={false}
          style={
                    Object {
                              "position": "fixed",
                              "right": "25%",
                              "top": "70px",
                            }
          }
/>,
        <RaisedButton
          disabled={false}
          fullWidth={false}
          id="home-button"
          label="Home"
          labelPosition="after"
          onClick={[Function]}
          primary={false}
          secondary={false}
          style={
                    Object {
                              "position": "fixed",
                              "right": "25%",
                              "top": "120px",
                            }
          }
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "disabled": false,
          "fullWidth": false,
          "id": "panic-button",
          "label": "PANIC",
          "labelPosition": "after",
          "onClick": [Function],
          "primary": false,
          "secondary": false,
          "style": Object {
            "position": "fixed",
            "right": "25%",
            "top": "20px",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Avatar
              alt="Your user photo"
              size={40}
              src="https://speakerdata2.s3.amazonaws.com/photo/image/839843/thats-what-she-said-0413-1-lgn.jpg"
/>,
            <h3>
              lolakoala
</h3>,
          ],
          "id": "current-user",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "alt": "Your user photo",
              "size": 40,
              "src": "https://speakerdata2.s3.amazonaws.com/photo/image/839843/thats-what-she-said-0413-1-lgn.jpg",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "lolakoala",
            },
            "ref": null,
            "rendered": "lolakoala",
            "type": "h3",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "logo",
          "id": "logo",
          "src": "logo1.png",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "disabled": false,
          "fullWidth": false,
          "href": "/",
          "label": "Sign Out",
          "labelPosition": "after",
          "linkButton": true,
          "onClick": [Function],
          "primary": false,
          "secondary": false,
          "style": Object {
            "position": "fixed",
            "right": "25%",
            "top": "70px",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "disabled": false,
          "fullWidth": false,
          "id": "home-button",
          "label": "Home",
          "labelPosition": "after",
          "onClick": [Function],
          "primary": false,
          "secondary": false,
          "style": Object {
            "position": "fixed",
            "right": "25%",
            "top": "120px",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <RaisedButton
            disabled={false}
            fullWidth={false}
            id="panic-button"
            label="PANIC"
            labelPosition="after"
            onClick={[Function]}
            primary={false}
            secondary={false}
            style={
                        Object {
                                    "position": "fixed",
                                    "right": "25%",
                                    "top": "20px",
                                  }
            }
/>,
          <div
            id="current-user"
>
            <Avatar
                        alt="Your user photo"
                        size={40}
                        src="https://speakerdata2.s3.amazonaws.com/photo/image/839843/thats-what-she-said-0413-1-lgn.jpg"
            />
            <h3>
                        lolakoala
            </h3>
</div>,
          <img
            alt="logo"
            id="logo"
            src="logo1.png"
/>,
          <RaisedButton
            disabled={false}
            fullWidth={false}
            href="/"
            label="Sign Out"
            labelPosition="after"
            linkButton={true}
            onClick={[Function]}
            primary={false}
            secondary={false}
            style={
                        Object {
                                    "position": "fixed",
                                    "right": "25%",
                                    "top": "70px",
                                  }
            }
/>,
          <RaisedButton
            disabled={false}
            fullWidth={false}
            id="home-button"
            label="Home"
            labelPosition="after"
            onClick={[Function]}
            primary={false}
            secondary={false}
            style={
                        Object {
                                    "position": "fixed",
                                    "right": "25%",
                                    "top": "120px",
                                  }
            }
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "disabled": false,
            "fullWidth": false,
            "id": "panic-button",
            "label": "PANIC",
            "labelPosition": "after",
            "onClick": [Function],
            "primary": false,
            "secondary": false,
            "style": Object {
              "position": "fixed",
              "right": "25%",
              "top": "20px",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Avatar
                alt="Your user photo"
                size={40}
                src="https://speakerdata2.s3.amazonaws.com/photo/image/839843/thats-what-she-said-0413-1-lgn.jpg"
/>,
              <h3>
                lolakoala
</h3>,
            ],
            "id": "current-user",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "alt": "Your user photo",
                "size": 40,
                "src": "https://speakerdata2.s3.amazonaws.com/photo/image/839843/thats-what-she-said-0413-1-lgn.jpg",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "lolakoala",
              },
              "ref": null,
              "rendered": "lolakoala",
              "type": "h3",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "logo",
            "id": "logo",
            "src": "logo1.png",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "disabled": false,
            "fullWidth": false,
            "href": "/",
            "label": "Sign Out",
            "labelPosition": "after",
            "linkButton": true,
            "onClick": [Function],
            "primary": false,
            "secondary": false,
            "style": Object {
              "position": "fixed",
              "right": "25%",
              "top": "70px",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "disabled": false,
            "fullWidth": false,
            "id": "home-button",
            "label": "Home",
            "labelPosition": "after",
            "onClick": [Function],
            "primary": false,
            "secondary": false,
            "style": Object {
              "position": "fixed",
              "right": "25%",
              "top": "120px",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
